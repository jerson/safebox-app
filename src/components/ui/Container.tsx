import * as React from 'react';
import {
  StyleProp,
  StyleSheet,
  View,
  ViewProps,
  ViewStyle
} from 'react-native';
import Colors from '../../modules/constants/Colors';

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.grey1,
    flex: 1
  } as ViewStyle
});

export interface ContainerProps extends ViewProps {
  style?: StyleProp<ViewStyle>;
}

export interface ContainerState {}

export default class Container extends React.Component<
  ContainerProps,
  ContainerState
> {
  render() {
    const { style, children, ...props } = this.props;
    return (
      <View style={[styles.container, style]} {...props}>
        {children}
      </View>
    );
  }
}
