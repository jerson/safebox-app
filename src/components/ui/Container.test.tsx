import React from 'react';
import * as renderer from 'react-test-renderer';
import Container, {ContainerProps} from './Container';
import {View} from 'react-native';

const props: ContainerProps = {
  style: {},
  children: <View />,
};

test('renders without crashing', () => {
  const snapshot = renderer.create(<Container {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<Container {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
