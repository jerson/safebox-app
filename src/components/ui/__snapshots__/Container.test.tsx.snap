// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<_class
  style={
    Array [
      Object {
        "backgroundColor": "#fafdff",
        "flex": 1,
      },
      Object {},
    ]
  }
/>
`;
